src/Engine/GameInstance.java

```diff
@@ -150,4 +154,39 @@ public void endGame()
       GameEventListener.unregisterEventListener(co);
     }
   }
+
+  /**
+   * Returns a count of the number of still-living human players in the game.
+   */
+  public int countHumanPlayers()
+  {
+    int humans = 0;
+
+    for( Commander co : commanders )
+    {
+      if( !co.isDefeated && !co.isAI() )
+      {
+        humans++;
+      }
+    }
+    return humans;
+  }
+
+  /**
+   * Returns the map owned by the first human Commander found.
+   * Intended to be used when there is only one human player.
+   */
+  public GameMap getHumanPlayerMap()
```

Comment:
I'm not sure this really belongs in `GameInstance`. I can see moving `countHumanPlayers` here--maybe--but this is pretty specific to the map/fog-drawing logic - I don't see it ever being needed for actual game stuff.

Reply:
That's a fair argument. Could go in MapController, I guess?

Reply:
MapController shouldn't be concerned with drawing logic. I thing this really belongs where it's actually used, back in `SpriteMapView`.